packages/ckeditor5-font-mapper/src/fontMapping/FontReplacer.ts

```diff
@@ -0,0 +1,174 @@
+import Text from "@ckeditor/ckeditor5-engine/src/view/text";
+import UpcastWriter from "@ckeditor/ckeditor5-engine/src/view/upcastwriter";
+import DocumentFragment from "@ckeditor/ckeditor5-engine/src/view/documentfragment";
+import FontMapping from "./FontMapping";
+import FontMappingRegistry from "./FontMappingRegistry";
+
+const FONT_FAMILY_PROPERTY_NAME = "font-family";
+
+/**
+ * Recursively replaces the characters in direct children and removes their font-family style property
+ * if a font mapping for the font-family exists.
+ * The documentFragment given as the first parameter of this function will be altered directly.
+ *
+ * @param documentFragment - the document fragment
+ * @param parentFontMapping - parent font mapping to be respected in calculation of the applied font mapping.
+ */
+export const replaceFontInDocumentFragment = (
+  documentFragment: DocumentFragment,
+  parentFontMapping?: FontMapping
+): void => {
+  const childElements = findChildren(documentFragment) as Array<Element>;
+
+  for (const child of childElements) {
+    // get the font mapping for the child element or use the one inherited from the parent:
+    const fontMapping = computeFontMappingForElement(child, parentFontMapping);
+    if (!fontMapping) {
+      replaceFontInDocumentFragment(child, undefined);
+      continue;
+    }
+    // A new element will be cloned and font mappings will be applied to its
+    // text node children:
+    const replacementElement = createAlteredElementClone(fontMapping, child);
+    if (replacementElement) {
+      const childIndex: number = documentFragment.getChildIndex(child);
+      documentFragment._removeChildren(childIndex, 1);
+      documentFragment._insertChild(childIndex, replacementElement);
+    }
+
+    replaceFontInDocumentFragment(child, fontMapping);
+  }
+};
+
+/**
+ * Returns all direct children of a document fragment.
+ * Only returns elements of type {@link Element}.
+ *
+ * @param documentFragment - the document fragment
+ * @returns an array of child elements
+ */
+const findChildren = (documentFragment: DocumentFragment): Array<Element> => {
+  const children: Array<Element> = Array.from<Node>(documentFragment.getChildren())
+    .filter((value) => value instanceof Element)
+    .map((value) => value as Element);
+  return children;
+};
+
+/**
+ * Returns a registered {@link FontMapping} based on the given element.
+ *
+ * A {@link FontMapping} will be looked up by first checking if a font-family exist
+ * on the element. If a font-family exists a font mapping for the font-family will be returned
+ * or undefined if no font mapping exists.
+ * If no font-family exists on the element the inherited font mapping will be returned.
+ *
+ * @param element - the element to inspect
+ * @param inheritedFontMapping - a font mapping inherited from parent hierarchy
+ *    or undefined if no font-family exist in the parent hierarchy
+ * @returns the fontMapping based on the given element or undefined
+ */
```

Comment:
Minor typos, slight reformat:

```suggestion
/**
 * Returns a registered {@link FontMapping} based on the given element.
 *
 * A {@link FontMapping} will be looked up by first checking if a font-family
 * exists on the element:
 *
 * * If a font-family exists, a font mapping for the font-family will be returned
 *   or undefined if no font mapping exists.
 *
 * * If no font-family exists, on the element the inherited font mapping will be
 *   returned.
 *
 * @param element - the element to inspect
 * @param inheritedFontMapping - a font mapping inherited from parent hierarchy
 *    or `undefined` if no font-family exist in the parent hierarchy
 * @returns the fontMapping based on the given element or `undefined`
 */
```